core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -119,11 +119,11 @@ object BattleDamage {
                 modifiers["Landing"] = -50
 
             // Land Melee Unit attacking to Water
-            if (attacker.unit.type.isLandUnit() && !attacker.unit.isEmbarked() && attacker.isMelee() && defender.getTile().isWater
+            if (attacker.unit.type.isLandUnit() && !attacker.getTile().isWater && attacker.isMelee() && defender.getTile().isWater
                     && !attacker.unit.hasUnique(UniqueType.AttackAcrossCoast))
                 modifiers["Boarding"] = -50
-            // Naval Unit Melee attacking to Land (not City) unit
-            if (attacker.unit.type.isWaterUnit() && attacker.isMelee() && !defender.getTile().isWater
+            // Melee Unit on water attacking to Land (not City) unit
+            if (!attacker.unit.type.isAirUnit() && attacker.isMelee() && attacker.getTile().isWater && !defender.getTile().isWater
```

Comment:
Or when attacking from water

Reply:
Both of these cases were to keep consistency with the code already here. I don't remember how Civ 5 handles Lekmod's helicopters, or even if it was added as a dll in Civ 5 (for reference, in Lekmod did something like this for helicopters a long while back). And given we already have a unique to remove this penalty, this seems like something that isn't as big of a deal as I may seem. In my eyes, that's more of a design question for us to ask (or poll on discord) than an actual problem